src/main/haskell/language-kore/test/parser/Data/Kore/Step/AxiomPatternsTest.hs

```diff
@@ -0,0 +1,172 @@
+module Data.Kore.Step.AxiomPatternsTest where
+
+import           Test.Tasty                                          (TestTree,
+                                                                      testGroup)
+import           Test.Tasty.HUnit                                    (assertEqual,
+                                                                      testCase)
+
+import           Data.Kore.AST.Builders
+import           Data.Kore.AST.Common
+import           Data.Kore.AST.MetaOrObject
+import           Data.Kore.AST.MLPatternsTest                        (extractPurePattern)
+import           Data.Kore.AST.PureML
+import           Data.Kore.AST.Sentence
+import           Data.Kore.ASTVerifier.DefinitionVerifierTestHelpers
+import           Data.Kore.Parser.ParserImpl
+import           Data.Kore.Parser.ParserUtils
+import           Data.Kore.Step.AxiomPatterns
+
+axiomPatternsTests :: TestTree
+axiomPatternsTests =
+    testGroup
+        "AxiomPatterns Tests"
+        [ axiomPatternsUnitTests
+        , axiomPatternsIntegrationTests
+        ]
+
+axiomPatternsUnitTests :: TestTree
+axiomPatternsUnitTests =
+    testGroup
+        "AxiomPatterns Unit Tests"
+        []
+
+axiomPatternsIntegrationTests :: TestTree
+axiomPatternsIntegrationTests =
+    testGroup
+        "AxiomPatterns Unit Tests"
+        [ testCase "I1 <= I2 => I1 <=Int I2"
+            (assertEqual
+                ""
+                (Right expectedAxiomPattern1)
+                (koreSentenceToAxiomPattern Object testAxiom1)
+            )
+
+        ]
+
+testAxiomString1 :: String
+testAxiomString1 =
+  "axiom{} \\and{TCell{}} (\n\
+  \  \\top{TCell{}}(), \\and{TCell{}} (\n\
+  \  \\top{TCell{}}(), \\rewrites{TCell{}}(\
+    \Lbl'-LT-'T'-GT-'{}(Lbl'-LT-'k'-GT-'{}(\
```

Comment:
I know that this is a real-life test, but I'd re-indent it and I'd rename things like `Lbl'-LT-'T'-GT-'` to `LT-T-GT` or even `T`.